developer-portal/for-developers/custom-integration/fast-sdks/ios/mobile-sdk.mdx

```diff
@@ -0,0 +1,219 @@
+button.addTarget(self, action: #selector(invokeFastCheckout), for: .touchUpInside)
+```
+
+## Use Fast Checkout
+
+<MobileInvokeCheckout
+  sdkModuleName="FastKit"
+  actionType="handler"
+  actionName="FastCheckoutButton"
+/>
+
+The SDK provides 2 main routes for invoking checkout:
+
+1. [Checkout with a List of Products](#checkout-with-a-list-of-products)
+2. [Checkout with a Cart ID](#checkout-with-a-cart-id)
+
+<MobileProductOptionsWarning sdkFullName="Fast iOS SDK" />
+
+### Checkout with a List of Products
+
+Pass in a list of products. A `Product` is a struct describing the item being ordered:
+
+```swift
+/// Present the Fast Checkout flow.
+/// - presentedBy: The presenting view controller for the Fast Checkout flow.
+/// - products: The list of products the user is ordering,
+/// - couponCode: A discount code that the user might have entered.
+/// - callback: A callback to listen for events during checkout.
+public func checkout(
+	presentedBy: UIViewController,
+	products: [Product],
+	couponCode: String,
+	callback: @escaping (FastEvent) -> Void
+)
+```
+
+#### Product Object Breakdown
+
+The `Product` object describes any product that can be ingested by the Fast backend and thus purchased via the SDK.
+
+| Name              | Type              | Description                                                       |
+| ----------------- | ----------------- | ----------------------------------------------------------------- |
+| platformProductId | String            | The product identifier.                                           |
+| platformVariantId | String?           | A unique sub-identifier for this product.                         |
+| productOptions    | [`ProductOption`] | Configurable properties of this Product (e.g. color, size, etc.). |
+| quantity          | Int               | The number of this item that should be purchased.                 |
+
+#### ProductOption Breakdown
+
+A `ProductOption` is a configuration that further describes the product being ordered (e.g. _ProductOption(id: "color", value: "blue"))._
+
+| Name       | Type   | Description                                            |
+| ---------- | ------ | ------------------------------------------------------ |
+| identifier | String | The identifier of the product option or configuration. |
+| value      | String | The value of the specified option.                     |
+
+### Checkout with a Cart ID
+
+Pass in a `cartId` that the Fast backend can break into a list of products and, if applicable, any discount code.
+
+```swift
+/// Present the Fast Checkout flow.
+/// - cartId: Unique identifier for a user's cart.
+/// - couponCode: A discount code that the user might have entered.
+/// - callback: A callback to listen for events during checkout.
+public func checkout(
+	presentedBy: UIViewController,
+	cartId: String,
+	couponCode: String,
+	callback: @escaping (FastEvent) -> Void
+)
+```
+
+### Initialize Theme
```

Comment:
I don't think this is something we want to expose on iOS.

Reply:
@tsabend 
Oh, so for Android we're doing: theme defaults to system theme but can be set manually to light or dark.

But for iOS... only inheriting system theme or host theme?